### DISCUSSION TITLE
Overlap between CSF mask and Brain template when using SIGMA Wistar atlas when preprocessing.

### DISCUSSION THREAD

**Discussion by KimMinhHuynh** [CoBrALab/RABIES#369] [Q&A] [answered]
Hi 

In singularity, I'm getting an overlap error between my mask and template during preprocess. I run this:

`singularity run -B $PWD/test_dataset:/test_dataset:ro -B $PWD/test_dataset_outputs:/test_dataset_outputs/ -B $PWD/ sigma_wistar/SIGMA_Version1.2.1:/atlas/ /home/sdc/Desktop/rabis.sif -p MultiProc \preprocess_test_dataset/ /test_dataset_outputs/ --commonspace_reg masking=true,brain_extraction=false,template_registration=Syn,fast_commonspace=false --anat_robust_inho_cor apply=true,masking=false,brain_extraction=false,template_registration=Syn --anat_template=/atlas/ SIGMA_Rat_Anatomical_Imaging/SIGMA_Rat_Anatomical_InVivo_Template/SIGMA_InVivo_Brain_Template.nii --brain_mask=/ atlas/SIGMA_Rat_Anatomical_Imaging/SIGMA_Rat_Anatomical_InVivo_Template/SIGMA_InVivo_Brain_Mask.nii --WM_mask=/ atlas/SIGMA_Rat_Anatomical_Imaging/SIGMA_Rat_Anatomical_InVivo_Template/SIGMA_InVivo_WM_Mask.nii --CSF_mask=/ atlas/SIGMA_Rat_Anatomical_Imaging/SIGMA_Rat_Anatomical_InVivo_Template/SIGMA_InVivo_CSF_Mask.nii --vascular_mask=/atlas/SIGMA_Rat_Anatomical_Imaging/SIGMA_Rat_Anatomical_InVivo_Template/SIGMA_InVivo_Brain_Mask.nii --labels=/atlas/SIGMA_Rat_Brain_Atlases/SIGMA_Anatomical_Atlas/InVivo_Atlas/ SIGMA_InVivo_Anatomical_Brain_Atlas.nii
`
and receive this:

![Screenshot 2024-05-10 152958](https://github.com/CoBrALab/RABIES/assets/43877678/e4f437ba-9c41-4a43-8e48-fbdc18f5591f)

I have tried without the atlas with no issue. Any suggestions on the cause?

**Comment by gdevenyi** [comment]
Please provide the complete logfile as well as some screenshots of the QC and input images you have.

**Comment by Gab-D-G** [comment, marked as answer]
Hi, this is the same as with #370 , you need to double check that the voxel/matrix dimensions as well as the image orientation match between your template files. Also make sure that the CSF/WM masks are binary masks. What happens if you replace the WM/CSF masks with your brain mask? does it run properly? (obviously then you won't be able to use the WM/CSF masks, but this would tell us that those files are the problem)

**Comment by KimMinhHuynh** [comment]
Sorry for not responding. Thank you @gdevenyi and @Gab-D-G for the help.
